implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/types/DynatraceSummary.java

```diff
@@ -15,61 +15,71 @@
  */
 package io.micrometer.dynatrace.types;
 
-import java.util.concurrent.atomic.AtomicLong;
-import java.util.concurrent.atomic.DoubleAdder;
-import java.util.concurrent.atomic.LongAdder;
-
 /**
  * Internal class for resettable summary statistics.
  *
  * @author Georg Pirklbauer
  */
 final class DynatraceSummary {
 
-    private final LongAdder count = new LongAdder();
+    private long count;
 
-    private final DoubleAdder total = new DoubleAdder();
+    private double total;
 
-    private final AtomicLong min = new AtomicLong();
+    private double min;
 
-    private final AtomicLong max = new AtomicLong();
+    private double max;
 
     void recordNonNegative(double amount) {
         if (amount < 0) {
             return;
         }
 
-        long longBits = Double.doubleToLongBits(amount);
         synchronized (this) {
-            max.getAndUpdate(prev -> Math.max(prev, longBits));
-            // have to check if a value was already recorded before, otherwise min will
-            // always stay 0 (because the default is 0).
-            min.getAndUpdate(prev -> count.longValue() > 0 ? Math.min(prev, longBits) : longBits);
-
-            total.add(amount);
-            count.increment();
+            max = Math.max(max, amount);
+            // check if count is equal to 0 and set initial min value if it is, otherwise
+            // min will always stay at 0.
+            min = count > 0 ? Math.min(min, amount) : amount;
+            total += amount;
+            count++;
         }
     }
 
+    /**
+     * Getters are not synchronized and might give inconsistent results. It is recommended
+     * to take a snapshot and use the {@link DynatraceSummarySnapshot} instead.
+     */
     long getCount() {
-        return count.longValue();
+        return count;
     }
 
+    /**
+     * Getters are not synchronized and might give inconsistent results. It is recommended
+     * to take a snapshot and use the {@link DynatraceSummarySnapshot} instead.
+     */
     double getTotal() {
-        return total.doubleValue();
+        return total;
     }
 
+    /**
+     * Getters are not synchronized and might give inconsistent results. It is recommended
+     * to take a snapshot and use the {@link DynatraceSummarySnapshot} instead.
+     */
     double getMin() {
-        return Double.longBitsToDouble(min.longValue());
+        return min;
     }
 
+    /**
+     * Getters are not synchronized and might give inconsistent results. It is recommended
+     * to take a snapshot and use the {@link DynatraceSummarySnapshot} instead.
+     */
     double getMax() {
-        return Double.longBitsToDouble(max.longValue());
+        return max;
```

Comment:
I added comments on the respective methods as well as on the methods that call these methods, to call this out explicitly. The methods on DynatraceTimer and DynatraceDistributionSummary should never be called, since we use the snapshotting approach to read the values.

Reply:
As an alternative, we could also throw an exception when the methods (min/max/total/count) on the DynatraceTimer / DynatraceDistributionSummary are called, but I wanted to refrain from changing the behavior greatly when applying a bugfix.

Reply:
The problem is that users will essentially never use the Dynatrace types directly. They'll use `Timer` or `DistributionSummary` which won't show them the JavaDocs on the overridden methods. It would be somewhat unusual for users to directly read from these in their applications if they're using the Dynatrace registry, also. That said, I guess the behavior is what's needed for Dynatrace. I'm fine ignoring this if you are.

Reply:
I assume users would never interact with the `DistributionSummary` methods directly if there is an exporter that does it for them. My thinking is, as the `DynatraceDistributionSummary` is only used in the `DynatraceExporter`, the Interface is used mainly as a contract, but not actively used in user code. Do you see a lot of direct use of these methods?

Reply:
@sonatype-lift ignoreall